editor/components/inserter/menu.js

```diff
@@ -119,140 +84,127 @@ class InserterMenu extends wp.element.Component {
 		)( blockTypes );
 	}
 
-	findNext( currentBlock, blockTypes ) {
-		/**
-		 * null is the value that will trigger iterating back to
-		 * the top of the list of block types.
-		 */
-		if ( null === currentBlock ) {
-			return blockTypes[ 0 ].slug;
-		}
+	findByIncrement( blockTypes, increment = 1 ) {
+		// Add on a fake search block to the list to cycle through.
+		const list = blockTypes.concat( { slug: 'search' } );
 
-		const currentIndex = blockTypes.findIndex( ( blockType ) => currentBlock === blockType.slug );
-		const nextIndex = currentIndex + 1;
-		const highestIndex = blockTypes.length - 1;
+		const currentIndex = findIndex( list, ( blockType ) => this.state.currentFocus === blockType.slug );
```

Comment:
No need to change, but Lodash's shorthands (which are unfortunately slated to be disabled-by-default in the [upcoming major version](https://github.com/lodash/lodash/wiki/Roadmap)) are a nice simplification.

Before:

```js
findIndex( list, ( blockType ) => this.state.currentFocus === blockType.slug )
```

After:

```js
findIndex( list, { slug: this.state.currentFocus } )
```